test/parallel/test-http2-stream-destroy-event-order.js

```diff
@@ -21,8 +21,8 @@ server.listen(0, common.mustCall(() => {
   client = http2.connect(`http://localhost:${server.address().port}`);
   req = client.request();
   req.resume();
-  req.on('close', common.mustCall(() => {
-    req.on('error', common.mustCall(() => {
+  req.on('error', common.mustCall(() => {
+    req.on('close', common.mustCall(() => {
```

Comment:
are these changes part of this PR, if so why?

Reply:
I messed up the emit order in my streams error handling PR (error should be before close). Otherwise these modules will report a premature close error instead of the error passed to stream.destroy. It's quite related to the end-of-stream/pump stuff so just bundled that in here. Want me to open an independent PR with that? (tests won't pass without this fix)

Reply:
yes please! If possible I would like to backport this.

Reply:
:+1: I don't think a backport is needed as the stream error handling stuff we did is going out in 10 only I think?

Reply:
I'm splitting it out in a sec

Reply:
https://github.com/nodejs/node/pull/19836